### PR TITLE
SWIFT-318 Write an error handling guide

### PR DESCRIPTION
**Pull request by patrickfreed**
This PR adds an error handling guide. It also creates a `Guides` directory and moves `Development.md` into it, along with the error one.

### CODE REVIEW

File: Guides/Error-Handling.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+**Error Labels:** Some types of errors may contain more specific information describing the context in which they occured. This information is conveyed through the usage of `errorLabels`. Specifically, any server error or connection related error may contain labels. They are primarily used for classifying errors that occur when performing transactions, so at this point they will largely be unused.
```
Comment (kmahar) [inline Guides/Error-Handling.md]: maybe leave off the last sentence, we will probably forget to remove it whenever we get to adding 4.0 support... lol. or maybe add a ticket to 4.0 epic telling us to update this guide to talk more about transaction errors? if you do leave it I would suggest stating explicitly why this is the case (no transactions support).

File: Guides/Error-Handling.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+
+## Examples
+### Handling any error thrown by the driver
+```
+do {
+    // something involving the driver
+} catch let error as MongoSwiftError {
+    print("Driver error!")
+    if let serverError = error as? ServerError { ... }
```
[the remark this answers is not in the document]
Comment (kmahar) [reply Guides/Error-Handling.md]: `MongoSwiftError` -> `MongoError`?

[the remark this answers is not in the document]
Comment (patrickfreed) [reply Guides/Error-Handling.md]: Yes! thanks for catching that.

File: Makefile [hunk 3]
```diff
@@ -43,4 +43,4 @@ clean:
 documentation:
 	make project
 	@$(call check_for_gem,jazzy)
-	jazzy --module MongoSwift --module-version 0.0.9 --root-url https://mongodb.github.io/mongo-swift-driver/ --documentation Development.md
+	jazzy --module MongoSwift --module-version 0.0.9 --root-url https://mongodb.github.io/mongo-swift-driver/ --documentation=Guides/*.md
```
Comment (mbroadst) [inline Makefile]: So _all_ generated documentation goes into `Guides` now?

Comment (patrickfreed) [reply Makefile]: No, the `--documentation` option just isn't named very well. It's used for specifying input files for guides. See the [jazzy docs](https://github.com/realm/jazzy#guides) for more info.

Comment (mbroadst) [reply Makefile]: d'oh, okay

**Pull request merged**